packages/realm-web/src/MongoDBRealmError.ts

```diff
@@ -49,31 +53,67 @@ export class MongoDBRealmError extends Error {
      */
     public readonly link: string | undefined;
 
+    /**
+     * Construct an error from a request and a response.
+     *
+     * @param request The request.
+     * @param response The response.
+     */
+    public static async fromRequestAndResponse(
+        request: Request<unknown>,
+        response: FetchResponse,
+    ): Promise<MongoDBRealmError> {
+        const { url, method } = request;
+        const { status, statusText } = response;
+        if (
+            response.headers.get("content-type")?.startsWith("application/json")
+        ) {
+            const body = await response.json();
+            const error = body.error || "No message";
+            const errorCode = body.error_code;
+            const link = body.link;
+            return new MongoDBRealmError(
+                method as Method,
+                url,
+                status,
+                statusText,
+                error,
+                errorCode,
+                link,
+            );
+        } else {
+            return new MongoDBRealmError(
+                method as Method,
+                url,
+                status,
+                statusText,
+            );
+        }
+    }
+
     constructor(
         method: Method,
         url: string,
         statusCode: number,
```

Comment:
Maybe not in this commit, but we may need to revisit this for watch since in-band errors in the stream will have an error, errorCode, and link, but won't have the http bits.